### PR TITLE
Feature/about version

### PR DESCRIPTION
**Pull request by Adham32**

### CODE REVIEW

**Review by TinasheMzondiwa — APPROVED** [review]

File: features/settings/src/main/java/com/cryart/sabbathschool/settings/SSSettingsActivity.kt [hunk 1]
```diff
@@ -51,7 +55,7 @@ class SSSettingsActivity : SSColorSchemeActivity() {
 
         supportFragmentManager.beginTransaction().replace(
             R.id.ss_settings_frame,
-            SSSettingsFragment()
+            SSSettingsFragment.newInstance(appConfig)
```
Comment (merlin-zaraza) [inline features/settings/src/main/java/com/cryart/sabbathschool/settings/SSSettingsActivity.kt:58]: You can inject appConfig inside `SSSettingsFragment` and not pass it as argument

File: app/src/main/java/com/cryart/sabbathschool/ui/about/SSAboutViewModel.kt [hunk 2]
```diff
@@ -30,6 +31,8 @@ class SSAboutViewModel(private val context: Context) {
 
     val colorPrimary: Int get() = context.colorPrimary
 
+    val versionInfo: String get() = "${context.getString(R.string.ss_settings_version)}: ${BuildConfig.VERSION_NAME}"
```
Comment (merlin-zaraza) [inline app/src/main/java/com/cryart/sabbathschool/ui/about/SSAboutViewModel.kt]: Not sure this is right way for RTL languages. It's better to create a formatted record in `string.xml`, like `<string name="ss_settings_version">Version: %1$s</string>`

File: features/settings/src/main/java/com/cryart/sabbathschool/settings/SSSettingsActivity.kt (hunk 1 above)
Comment (TinasheMzondiwa) [reply features/settings/src/main/java/com/cryart/sabbathschool/settings/SSSettingsActivity.kt:58]: Unfortunately hilt cannot inject into a `PreferenceFragmentCompat`

Comment (merlin-zaraza) [reply features/settings/src/main/java/com/cryart/sabbathschool/settings/SSSettingsActivity.kt:58]: We can use a workaround described in [https://github.com/google/dagger/issues/2175](https://github.com/google/dagger/issues/2175). But, yeah, agree to leave it as is.

File: app/src/main/java/com/cryart/sabbathschool/ui/about/SSAboutViewModel.kt (hunk 2 above)
Comment (Adham32) [reply app/src/main/java/com/cryart/sabbathschool/ui/about/SSAboutViewModel.kt]: updated and fixed ;-)

File: common/translations/src/main/res/values-ar/strings.xml [hunk 3]
```diff
@@ -97,4 +98,5 @@
     <string name="ss_reading_offline_retry">"أعد المحاولة"</string>
     <string name="ss_lessons_previous">الدروس السابقة</string>
     <string name="ss_about_text">مهمة Adventech هي أولاً وقبل كل شيء - أعط المجد لربنا ، أيام الأيام القديمة. نحن ملتزمون بالقيام بوزارتنا من خلال التكنولوجيا. هدفنا هو توحيد كنيستنا السبتية. هدفنا هو الإعلان عن أخبار جيدة لهذا العالم من خلال التكنولوجيا والتقدم في الاتصالات. هدفنا هو رؤية المجيء الثاني ليسوع.  لقد ألهمنا مثال يسوع وأساليبه عندما خدم هنا على الأرض. لذلك ، نحن ملتزمون بتقديم درجة عالية من الجودة لجميع مشاريعنا دون أي تكلفة على الإطلاق مع الحب والتفاني. نحن ملتزمون بإعطاء المجد لأبينا ، لأنه يستحق أعلى الثناء.  نحن نؤمن حقًا أن الله قد زاد معرفة العالم الحديث ويريدنا أن نستخدمه لمجده لإعلان عودته قريبًا!</string>
+    <string name="ss_settings_version_with_param_string">الإصدار: %1$S</string>
```
Comment (TinasheMzondiwa) [inline common/translations/src/main/res/values-ar/strings.xml:101]: This does not look correct

File: common/translations/src/main/res/values-fa/strings.xml [hunk 4]
```diff
@@ -103,4 +104,5 @@
 
     <!-- Previous lessons -->
     <string name="ss_lessons_previous">"دروس قبلی"</string>
+    <string name="ss_settings_version_with_param_string">نسخه: %1$S</string>
```
Comment (TinasheMzondiwa) [inline common/translations/src/main/res/values-fa/strings.xml:107]: Here as well

File: common/translations/src/main/res/values-iw/strings.xml [hunk 5]
```diff
@@ -103,4 +104,5 @@
 
     <!-- Previous lessons -->
     <string name="ss_lessons_previous">"שיעורים קודמים"</string>
+    <string name="ss_settings_version_with_param_string">גִרְסָה: %1$S</string>
```
Comment (TinasheMzondiwa) [inline common/translations/src/main/res/values-iw/strings.xml:107]: Will this work?

**Review by TinasheMzondiwa — CHANGES_REQUESTED** [review]
Requires an update

File: common/translations/src/main/res/values-iw/strings.xml (hunk 5 above)
Comment (Adham32) [reply common/translations/src/main/res/values-iw/strings.xml:107]: In Android Studio I have:
`<string name="ss_settings_version_with_param_string">Sنسخه: %$1</string>`

**Review by TinasheMzondiwa — APPROVED** [review]

**Pull request merged**